src/StakeRegistry.sol

```diff
@@ -283,71 +251,111 @@ contract StakeRegistry is VoteWeigherBase, StakeRegistryStorage {
+        if (currentStake < minimumStakeForQuorum[quorumNumber]) {
+            currentStake = uint96(0);
         }
-        // get stakeBeforeUpdate and update with new stake
-        uint96 stakeBeforeUpdate = _recordOperatorStakeUpdate({
+        
+        // Update the operator's stake and retrieve the delta
+        int256 delta = _recordOperatorStakeUpdate({
             operatorId: operatorId, 
             quorumNumber: quorumNumber, 
-            operatorStakeUpdate: operatorStakeUpdate
+            newStake: currentStake
         });
 
-        emit StakeUpdate(operatorId, quorumNumber, operatorStakeUpdate.stake);
-
-        return (stakeBeforeUpdate, operatorStakeUpdate.stake);
+        return (delta, currentStake);
     }
 
     /**
      * @notice Records that `operatorId`'s current stake for `quorumNumber` is now param @operatorStakeUpdate
-     * and returns the previous stake.
+     * @return The change in the operator's stake as a signed int256
      */
     function _recordOperatorStakeUpdate(
         bytes32 operatorId,
         uint8 quorumNumber,
-        OperatorStakeUpdate memory operatorStakeUpdate
-    ) internal returns (uint96) {
-        // initialize stakeBeforeUpdate to 0
-        uint96 stakeBeforeUpdate;
-        uint256 operatorStakeHistoryLength = operatorIdToStakeHistory[operatorId][quorumNumber].length;
-
-        if (operatorStakeHistoryLength != 0) {
-            // set nextUpdateBlockNumber in prev stakes
-            operatorIdToStakeHistory[operatorId][quorumNumber][operatorStakeHistoryLength - 1]
+        uint96 newStake
+    ) internal returns (int256) {
+        /**
+         * If the operator has previous stake history, update the previous entry
+         * and fetch their previous stake
+         */
+        uint96 prevStake;
+        uint256 historyLength = operatorIdToStakeHistory[operatorId][quorumNumber].length;
+        if (historyLength != 0) {
+            operatorIdToStakeHistory[operatorId][quorumNumber][historyLength - 1]
                 .nextUpdateBlockNumber = uint32(block.number);
-            // load stake before update into memory if it exists
-            stakeBeforeUpdate = operatorIdToStakeHistory[operatorId][quorumNumber][operatorStakeHistoryLength - 1]
-                .stake;
+
+            prevStake = 
+                operatorIdToStakeHistory[operatorId][quorumNumber][historyLength - 1].stake;
         }
-        // push new stake to storage
-        operatorIdToStakeHistory[operatorId][quorumNumber].push(operatorStakeUpdate);
-        return stakeBeforeUpdate;
+        
+        // Create a new stake update and push it to storage
+        // TODO - update the entry instead of pushing, if the last update block is this block
+        operatorIdToStakeHistory[operatorId][quorumNumber].push(OperatorStakeUpdate({
+            updateBlockNumber: uint32(block.number),
+            nextUpdateBlockNumber: 0,
+            stake: newStake
+        }));
```

Comment:
Question for reviewers - currently, our history gets pushed to every time we do an update, even if the last update was the same block. 

This can lead to chains in the history where multiple history updates that occur in the same block have a "nextUpdateBlockNum" field that points to the same block as the next update.

I feel like it makes more sense to just check if the last update is the current block, and update that entry's stake if so.

Reply:
that would be a great change! i often need to take a second to remember how to handle the case of many updates in the same block